src/lib/optimizer/strategy/null_scan_removal_rule.cpp

```diff
@@ -0,0 +1,84 @@
+#include "null_scan_removal_rule.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "constant_mappings.hpp"
+#include "cost_estimation/abstract_cost_estimator.hpp"
+#include "expression/is_null_expression.hpp"
+#include "expression/lqp_column_expression.hpp"
+#include "hyrise.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "optimizer/join_ordering/join_graph.hpp"
+#include "statistics/cardinality_estimation_cache.hpp"
+#include "statistics/cardinality_estimator.hpp"
+#include "statistics/table_statistics.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+void NullScanRemovalRule::apply_to_plan(const std::shared_ptr<LogicalPlanRootNode>& root) const {
+  Assert(root->type == LQPNodeType::Root, "NullScanRemovalRule needs root to hold onto");
+
+  std::vector<std::shared_ptr<AbstractLQPNode>> nodes_to_remove;
+
+  // To determine if the rule applies to a node it must meet all of the following conditions:
+  // 1. The node must be of type Predicate
+  // 2. The predicate must be a `null expression`
+  // 3. The predicate condition must be `is not null`
+  // 4. The predicate operand needs to be an `LQP Column expression`
```

Comment:
Backticks are reserved for code (same as in markdown). Just use the `needs to be an LQPColumnExpression`?